src/averageAges.js

```diff
@@ -55,7 +77,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person => person.sex === 'f')
+    .filter(mother =>
+      people.some(person => person.mother === mother.name));
+
+  let children = people.filter(person =>
+    mothers.some(mother => person.mother === mother.name));
+
+  if (onlyWithSon) {
+    children = children.filter(child => child.sex === 'm');
+  }
+
+  const ageEachDiff = children.map(child =>
+    child.born - mothers.find(mother => mother.name === child.mother).born);
+
+  const diffAgeSum = ageEachDiff
+    .reduce((sumOfAge, prevSum) => sumOfAge + prevSum);
+
+  return diffAgeSum / ageEachDiff.length;
```

Comment:
You use `reduce` and division by length in every function. Maybe we should move this logic to a separate helper?